db-client/src/main/java/com/splicemachine/db/client/am/Statement.java

```diff
@@ -806,8 +806,16 @@ public void cancel() throws SQLException {
                 agent_.logWriter_.traceEntry(this, "cancel");
             }
             checkForClosedStatement(); // Per jdbc spec (see java.sql.Statement.close() javadoc)
-            throw new SqlException(agent_.logWriter_, 
-                new ClientMessageId(SQLState.CANCEL_NOT_SUPPORTED_BY_SERVER));
+            byte[] token = connection_.getInterruptToken();
+            if (token == null) {
+                throw new SqlException(agent_.logWriter_,
+                        new ClientMessageId(SQLState.CANCEL_NOT_SUPPORTED_BY_SERVER));
+            }
+            InterruptionToken it = new InterruptionToken(token);
+            Connection sideConnection = connection_.getSideConnection();
+            try (java.sql.Statement s = sideConnection.createStatement()) {
+                s.execute("call SYSCS_UTIL.SYSCS_KILL_DRDA_OPERATION('" + it.toString() +"')");
```

Comment:
Why not `Connection.prepareStatement` with the `it.toString` as a parameter? I'm hoping we could avoid any string concatenation for SQL queries.

I'd also make the `call ...` a constant (a final value) and ultimately a configuration (not a hardcoded value).

Reply:
I'm trying to avoid a second roundtrip for a path that won't benefit too much from using a prepared statement. InterruptionToken is sent by the server so I'm not worried about a SQL injection here.